internal/controller/management/serviceaccounts/serviceaccounts.go

```diff
@@ -0,0 +1,113 @@
+package serviceaccounts
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	rbacapi "github.com/akuity/kargo/api/rbac/v1alpha1"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+// Old annotation keys
+const (
+	OldAnnotationKeySub    = "rbac.kargo.akuity.io/sub"
+	OldAnnotationKeyEmail  = "rbac.kargo.akuity.io/email"
+	OldAnnotationKeyGroups = "rbac.kargo.akuity.io/groups"
+)
+
+// ServiceAccountReconciler Reconciles for ServiceAccounts that need annotation migration
+type ServiceAccountReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Logger *logging.Logger
```

Comment:
Let's not have `Logger` be a field. We can use `logging.LoggerFromContext(ctx)` to pluck a properly configured logger out of thin air wherever we need it. (And often the context-bound logger has had additional details added to it by some caller, so our logs tend to be more information-rich if we stick to using that.)